src/main/java/com/google/api/generator/gapic/composer/SampleCodeHelperComposer.java

```diff
@@ -284,13 +293,46 @@ private static TryCatchStatement composeStreamRpcCallableMethodSampleCode(
   }
 
   private static TryCatchStatement composeStreamClientRpcCallableMethodSampleCode(
-      Method method, TypeNode clientType) {
+      Method method, TypeNode clientType, Map<String, ResourceName> resourceNames) {
+    // TODO(summerji): Add unit tests
+    // TODO(summerji): Add customize escape for @ character in <pre>{@code..}</pre>
+    // If variant method signatures exists, use the first one.
+    List<MethodArgument> arguments =
+        method.methodSignatures().isEmpty()
+            ? Collections.emptyList()
+            : method.methodSignatures().get(0);
+    MethodInvocationExpr onNextRequestMethodExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(createStreamObserverVarExpr(method, false))
+            .setMethodName("onNext")
+            .setArguments(createVariableExpr(REQUEST_VAR_NAME, method.inputType()))
+            .build();
+
+    List<Statement> bodyStatements = new ArrayList<>();
+    // Create StreamObserver response variable with anonymous class.
+    bodyStatements.add(
+        ExprStatement.withExpr(createAssignStreamObserverResponseAnonymousClassExpr(method)));
```

Comment:
This is currently creating the variable twice: Once here, and once in wcreateAssignStreamObserverRequestExpr`. In general, we should be treating `VariableExpr`s like real, Java variables. Could we refactor this so that it is only created once, and passed around?

Reply:
Also, PTAL at other such issues in this PR, and the other PRs as well. Will look at the downstream PRs when this has been addressed.